test/DefaultCluster.Tests/LifecycleObserverCreationTests.cs

```diff
@@ -0,0 +1,159 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Orleans.Runtime;
+using Orleans.TestingHost;
+using Orleans.TestingHost.Utils;
+using UnitTests.GrainInterfaces;
+using Xunit;
+
+namespace DefaultCluster.Tests.General
+{
+    /// <summary>
+    /// Tests for creating grain observers during lifecycle participation.
+    /// This validates that grain observers can be created during silo lifecycle stages,
+    /// which previously failed because lifecycle events were executed on a SystemTarget
+    /// (grain context), preventing CreateObjectReference from working.
+    /// </summary>
+    [TestCategory("BVT"), TestCategory("Lifecycle"), TestCategory("Observer")]
+    public class LifecycleObserverCreationTests : IClassFixture<LifecycleObserverCreationTests.Fixture>
+    {
+        private readonly IHost _host;
+
+        public class Fixture : IAsyncLifetime
+        {
+            private readonly TestClusterPortAllocator portAllocator;
```

Comment:
Use `InProcessTestClusterBuilder` so that you don't need to have a static `ObserverCreated` field. Instead of using a fixture, just create the cluster in-line in the test and have a single test method which checks everything we want to check.